File: frontend/src/utils/money.ts
/** «1 000 ₽», «833,33 ₽» — копейки показываются только когда они есть (доли поровну редко круглые). */
export function formatRub(kopecks: number): string {
  const sign = kopecks < 0 ? '−' : '';
  const abs = Math.abs(kopecks);
  const rub = Math.floor(abs / 100).toLocaleString('ru-RU');
  const kop = abs % 100;
  return kop === 0 ? `${sign}${rub} ₽` : `${sign}${rub},${String(kop).padStart(2, '0')} ₽`;
}

/** Ввод рублей из формы → копейки; null, если не число или ≤ 0. */
export function rubToKopecks(rub: string): number | null {
  const v = Number(rub.replace(',', '.').trim());
  if (!Number.isFinite(v) || v <= 0) return null;
  return Math.round(v * 100);
}

File: backend/src/main/kotlin/com/clubs/common/util/Money.kt
package com.clubs.common.util

import kotlin.math.abs

/** Суммы в DM и чат-постах: «1 000 ₽», «833,33 ₽». Копейки показываются только когда они есть. */
object Money {

    // Обычный пробел между разрядами: суммы короткие, а неразрывный ломал бы поиск по логам и тестам.
    private const val THOUSANDS_SEPARATOR = ' '

    fun rub(kopecks: Long): String {
        val sign = if (kopecks < 0) "−" else ""
        val rubles = abs(kopecks) / 100
        val kop = abs(kopecks) % 100
        val grouped = rubles.toString().reversed().chunked(3).joinToString(THOUSANDS_SEPARATOR.toString()).reversed()
        return if (kop == 0L) "$sign$grouped ₽" else "$sign$grouped,${kop.toString().padStart(2, '0')} ₽"
    }
}
